### ISSUE TITLE
speech-to-text container fails transcription after silence with error "Due to service inactivity, the client buffer exceeded maximum size."

### ISSUE THREAD

**Issue by Mikael-Gustavsson** [Azure-Samples/cognitive-services-speech-sdk#2234]
**Description**
The transcriptions is aborted when starting to speak again after a minute of silence. The error is 100% reproducable given the correct environmental settings and example file provided below.
The error is sporadic on production workloads (est 10% of production files, typically hour-long-interviews, are affected), but when they occurr, the same file will fail at the same spot all the time.

I am unsure if this is an error or a designed silence-timeout-abort. But since the error seems to be environment-dependent and the speech container is running on onprem HW and using standard components, it is very unexpected behaviour. If intended, I would like to be able to tweak the settings causing this behavior, or the behavior needs to be better described.

**IMPORTANT ISSUE UPDATE :**
Re-tested using SPX v1.34.1 on W11 with container running on WSL-based docker desktop, 
- Issue is _not_ present in SPX - Azure Speech CLI, Version 1.32.1
- Issue _is_ present in SPX - Azure Speech CLI, Version 1.34.1+6c915fe8d23c96b076ee0bb46877e982e2e516e6

**How to fail to reproduce:**
Run the test in docker on a client machine running w11. Transcribe completes correctly 3/3 times tested. (using SPX - Azure Speech CLI, Version 1.32.1)

**To Reproduce**
Run the test on a virtualized WS2022 in a (Vmware?) cluster with the transcription container running on docker in WSL on Ubuntu 22.04.
Yeah, its virtualization inception, but still, transcription fails 100% of times tested (more than 10).

See [Talk25s_silent60s_talk35s_Transcribe_Error.zip](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/14024383/Talk25s_silent60s_talk35s_Transcribe_Error.zip) for logs, and repro audio file.

1. In WS2022 WSL Start the speech-to-text container with the settings below:
        
        sudo docker run --rm -it -p 5000:5000 --memory 8g --cpus 4 mcr.microsoft.com/azure-cognitive-services/speechservices/speech-to-text:4.5.0-amd64-sv-se Eula=accept Billing=https://swedencentral.api.cognitive.microsoft.com/sts/v1.0/issuetoken ApiKey=5xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxa

3. On the WS02022 host, execute command:
        
        spx recognize --host ws://localhost:5000/ --file Talk25s_silent60s_talk35s_Transcribe_Error.wav --key none --region swedencentral --language sv-SE --output vtt file out.vtt

4. Wait for transcription to reach just past the one minute silence, transcription text appears in the console again, but aborts a few seconds later with the following console output

        RECOGNIZING: och så där ibland har det ju gått ungefär en minut av tystnad som ni fortsätter jag prata en
        CANCELED: Reason=Error
        CANCELED: ErrorCode=ServiceTimeout
        CANCELED: ErrorDetails=Due to service inactivity, the client buffer exceeded maximum size. Resetting the buffer. SessionId: 
        b6be3c97110c4bd6a6d8bc2dde5578a2
        CANCELED: Did you update the subscription info?
    
        SESSION STOPPED: b6be3c97110c4bd6a6d8bc2dde5578a2
        RECOGNIZING: och så där ibland har det ju gått ungefär en minut av tystnad som ni fortsätter jag prata en liten
        RECOGNIZING: och så där ibland har det ju gått ungefär en minut av tystnad som ni fortsätter jag prata en liten stund
        RECOGNIZED: Och så där ibland har det ju gått ungefär en minut av tystnad som ni fortsätter Jag prata en liten stund.
        
        PS E:\Transcribe\Testdata\errortest2> 

**Expected behavior:**
Transcription should complete correctly

**Additional context**
The issue is tested and present in the free, and sandard tier.
Billing is active and continousliy logged to the Azure portal (even the failed transcriptions are billed) 
The same error occurrs when calling the speech-to-text container from another container, using powershell core and spx, as well.
Other workloads running on the same container at the same time is not affected.
The speech-to-text container continues to run after the transcription error occurrs and can accept other workloads with no issues.

**Version of the Cognitive Services Speech SDK**
1. SPX - Azure Speech CLI, Version 1.34.1+6c915fe8d23c96b076ee0bb46877e982e2e516e6
2. mcr.microsoft.com/azure-cognitive-services/speechservices/speech-to-text:4.5.0-amd64-sv-se

**Platform, Operating System, and Programming Language**
1. Speech to text container: virtualized WS2022 in a (Vmware?) cluster with the transcription container running on docker in WSL on Ubuntu 22.04.
2. SPX can be run either from within another container or on the host.

**Detailed environment instructions**
I am unsure how much of this is required.
- Allow WSL on WS2022 VM

        Set-VMProcessor [VMName] -ExposeVirtualizationExtensions $true
- Install WSL

       Enable-WindowsOptionalFeature -Online -FeatureName Microsoft-Windows-Subsystem-Linux
       wsl --install
       wsl.exe --install -d ubuntu-22.04
- Update WSL distribution with docker-settings

       sudo apt-get update
       sudo apt-get install ca-certificates curl gnupg lsb-release
       sudo mkdir -p /etc/apt/keyrings
       curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
       echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu \
       $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
       cat /etc/apt/sources.list.d/docker.list
       sudo apt-get install docker-ce docker-ce-cli containerd.io docker-compose-plugin
       sudo service docker start

**Comment by Mikael-Gustavsson** [comment]
Note: see important issue update above, this seems to be an SPX version issue. (which I would guess extends to the SDK.

**Suggested Workaround**
Downgrade SPX to version 1.32.1
        
        dotnet tool uninstall --global Microsoft.CognitiveServices.Speech.CLI
        dotnet tool install --global Microsoft.CognitiveServices.Speech.CLI --version 1.32.1

**Comment by ralph-msft** [comment]
Thank you for using the cognitive speech services, and in particular thanks for the detailed and extremely helpful write up.

We have been able to reproduce the issue you are seeing. This is unfortunately a quirk with the recognition service hosted in the docker image not working correctly the newer configuration in `spx` from version 1.34.0 and newer. 

Running the latest version of the `spx` tool against the live production service does not reproduce this behaviour.

We've filed a bug to track fixing this issue internally in our backlog. We will respond here once we have an update.

In the meantime, there is another workaround you can use if you'd like to use the newest version of the spx tool:

1. Create a file called `speech.context.json` in the same directory that you'd like to run the spx tool from containing the following:
   
   ```json
   {"phraseDetection":{"language":"sv-SE"}}
   ```
   Please remember to update the language here to match your audio language

2. Add the following arguments to your spx command:

   ```
   --speech context @speech.context.json
   ```

Bug 6353256

**Comment by pankopon** [comment]
Original bug superseded by service incident ref. 470782607.

**Comment by ralph-msft** [comment]
Quick update: This issue is still being investigated and we currently do not have any ETA to share for a fix at this time.

**Comment by Mikael-Gustavsson** [comment]
**Update:** Issue seems to have been fixed in SPX somewhere in between versions 1.34 and 1.37.
Retested with SPX version 1,37 and 1.38, as well as transcribe container versions 1.4 and 1.7 and could not reproduce the issue using those versions. **I consider this issue RESOLVED**

**Comment by pankopon** [comment]
Closed as resolved based on customer update.
